pywr/solvers/glpk.pxi

```diff
@@ -46,6 +47,14 @@ cdef int term_hook(void *info, const char *s):
     return 1
 
 
+class GLPKError(Exception):
+    pass
+
+
+cdef void error_hook(void *info):
+    longjmp((<jmp_buf*>info)[0], <int>1)
```

Comment:
GLPK example calls `glp_free_env` here.

Reply:
Good catch. It's quite explicit that this needs to be done (before longjmp!)